### DISCUSSION TITLE
Lazy-loading race condition with select backends

### DISCUSSION THREAD

**Discussion by baodrate** [stevearc/dressing.nvim#91] [Q&A]
I've installed dressing using lazy.nvim. Dressing, telescope, and fzf-lua are all configured to lazy-load. I seem to be hitting a race condition where the vim.ui.select prompt falls back to the `builtin` backend on some runs of nvim. My uninformed guess is that the check for backend existence happens as soon as the dressing module is loaded.

Is there a way I can manually trigger the backend check? That way I can make sure it's updated once the backend has loaded.

**Comment by stevearc** [comment]
We actually don't do backend detection until you call `vim.ui.select` ([seen here](https://github.com/stevearc/dressing.nvim/blob/5f44f829481640be0f96759c965ae22a3bcaf7ce/lua/dressing/select/init.lua#L67)). If you follow that code, it's going to call [is_supported](https://github.com/stevearc/dressing.nvim/blob/5f44f829481640be0f96759c965ae22a3bcaf7ce/lua/dressing/select/init.lua#L11) which is just safely requiring telescope
https://github.com/stevearc/dressing.nvim/blob/5f44f829481640be0f96759c965ae22a3bcaf7ce/lua/dressing/select/telescope.lua#L3-L5
or fzf-lua
https://github.com/stevearc/dressing.nvim/blob/5f44f829481640be0f96759c965ae22a3bcaf7ce/lua/dressing/select/fzf_lua.lua#L3-L5

I'm not sure why it wouldn't be working, since lazy.nvim is supposed to automatically load the plugin when it's required. You could try adding some logging to these lines and see where exactly it's going wrong

**Reply by baodrate** [reply]
thank you for the info, that's very helpful. I'll try to investigate what's going on
